Storage/tests/Unit/StorageObjectTest.php

```diff
@@ -466,6 +466,26 @@ public function testDownloadsToFile()
         $this->assertEquals($string, $contents);
     }
 
+    public function testDownloadsToFileShouldNotCreateFileWhenObjectNotFound()
```

Comment:
This would be more valuable as a system test given that this is an issue in a way that the logic was implemented.

Reply:
The test has completely mocked out all network calls, still want me to move to System tests?